### PR TITLE
Added create, update and createIfNotExist operations

### PR DESCRIPTION
**Pull request by gandalfSievers**
Addressing #11232 added additional Create, CreateIfNotExist an Update Operations.

### CODE REVIEW

**Comment by acwtools** [comment]
Hi,

i don't know why travis-ci failed with PHP 5.4 - the failed test has no shared code with my changes and it also passes in my fork. I also created a VM with PHP 5.4 were i cannot reproduce it. Any advises?
Is there any chance to merge this PR?

Best regards

**Comment by Green-Cat** [comment]
Try to just rerun the tests on travis by closing and reopening the pull request.
I don't see any reason that specific test failed.

File: phalcon/mvc/collection.zep [hunk 1]
```diff
@@ -912,7 +924,205 @@ abstract class Collection implements EntityInterface, CollectionInterface, Injec
...
+					if exists === false {
+						if fetch id, data["_id"] {
+							let this->_id = id;
+						}
+					}
+				}
+			}
+		} else {
+			let success = false;
+		}
```
Comment (sergeyklay) [inline phalcon/mvc/collection.zep]: else not needed here

File: phalcon/mvc/collection.zep [hunk 2]
```diff
@@ -912,7 +924,205 @@ abstract class Collection implements EntityInterface, CollectionInterface, Injec
...
+		let status = collection->update(["_id": $this->_id], data, ["w": true]);
+		if typeof status == "array" {
+			if fetch ok, status["ok"] {
+				if ok {
+					let success = true;
+				}
+			}
+		} else {
+			let success = false;
+		}
```
Comment (sergeyklay) [inline phalcon/mvc/collection.zep]: else not needed here

**Comment by sergeyklay** [comment]
Update please `CHANGELOG.md`

**Comment by sergeyklay** [comment]
Can you please rebase?

**Comment by acwtools** [comment]
Im not sure if rebasing went wrong, if that is the case, i will redo the PR.

**Comment by sergeyklay** [comment]
Firstly you need submit just the commits that belong to the PR? Obviously you should rebase onto `2.1.x` branch. Then squash please commits into one to keep history clean.

**Comment by acwtools** [comment]
getting closer :) Sorry i lack of experience with github, i'm primarily working with closed source repos.

**Comment by acwtools** [comment]
Hmm now it must be ok.

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**